modules/custom/az_card/src/Plugin/Field/FieldWidget/AZCardWidget.php

```diff
@@ -20,11 +23,149 @@
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    $instance = parent::create(
+      $container,
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+    );
+
+    $instance->imageFactory = ($container->get('image.factory'));
+    $instance->pathValidator = ($container->get('path.validator'));
+    $instance->entityTypeManager = $container->get('entity_type.manager');
+    return $instance;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function form(FieldItemListInterface $items, array &$form, FormStateInterface $form_state, $get_delta = NULL) {
+
+    // Create shared settings for widget elemeents.
+    // This is necessary because wigets have to be AJAX replaced together,
+    // And in general we need a place to store shared settings.
+    $wrapper_id = Html::getUniqueId('az-card-wrapper');
+    $field_name = $this->fieldDefinition->getName();
+    $field_parents = $form['#parents'];
+    $field_state = static::getWidgetState($field_parents, $field_name, $form_state);
+    $field_state['ajax_wrapper_id'] = $wrapper_id;
+    $field_state['items_count'] = count($items);
+    $field_state['array_parents'] = [];
+    if (empty($field_state['open_status'])) {
+      $field_state['open_status'] = [];
+    }
+
+    // Persist the widget state so formElement() can access it.
+    static::setWidgetState($field_parents, $field_name, $form_state, $field_state);
+
+    $container = parent::form($items, $form, $form_state, $get_delta);
+    $container['widget']['#prefix'] = '<div id="' . $wrapper_id . '">';
+    $container['widget']['#suffix'] = '</div>';
+
+    if (isset($container['widget']['add_more']['#ajax']['wrapper'])) {
+      $container['widget']['add_more']['#ajax']['wrapper'] = $wrapper_id;
+    }
+    return $container;
+  }
+
   /**
    * {@inheritdoc}
    */
   public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state) {
 
+    // Get current collapse status.
+    $field_name = $this->fieldDefinition->getName();
+    $field_parents = $element['#field_parents'];
+    $widget_state = static::getWidgetState($field_parents, $field_name, $form_state);
+    $wrapper = $widget_state['ajax_wrapper_id'];
+    $status = (isset($widget_state['open_status'][$delta])) ? $widget_state['open_status'][$delta] : FALSE;
+    // New field values shouldn't be considered collapsed.
+    if ($items[$delta]->isEmpty()) {
+      $status = TRUE;
+    }
+
+    // Generate a preview if we need one.
+    if (!$status) {
+
+      // Bootstrap wrapper.
+      $element['preview_container'] = [
+        '#type' => 'container',
+        '#attributes' => [
+          'class' => ['col-12', 'col-sm-6', 'col-md-6', 'col-lg-4', 'card-preview',
+          ],
```

Comment:
Nitpick: is this closing bracket intentionally on a new line?

Reply:
I'll take a look at tweaking that - that one required some tinkering to get something `phpcs` was happy with. I do think it'd look better on the same line too.

Reply:
I think this should be resolved now.